RFS/src/main/java/com/rfs/RfsMigrateMetadata.java

```diff
@@ -0,0 +1,130 @@
+package com.rfs;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+
+import com.rfs.cms.CmsClient;
+import com.rfs.cms.OpenSearchCmsClient;
+import com.rfs.common.ClusterVersion;
+import com.rfs.common.ConnectionDetails;
+import com.rfs.common.GlobalMetadata;
+import com.rfs.common.IndexMetadata;
+import com.rfs.common.Logging;
+import com.rfs.common.OpenSearchClient;
+import com.rfs.common.S3Uri;
+import com.rfs.common.S3Repo;
+import com.rfs.common.SourceRepo;
+import com.rfs.common.TryHandlePhaseFailure;
+import com.rfs.common.SnapshotRepo;
+import com.rfs.transformers.TransformFunctions;
+import com.rfs.transformers.Transformer;
+import com.rfs.version_es_7_10.GlobalMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.IndexMetadataFactory_ES_7_10;
+import com.rfs.version_es_7_10.SnapshotRepoProvider_ES_7_10;
+import com.rfs.version_os_2_11.GlobalMetadataCreator_OS_2_11;
+import com.rfs.version_os_2_11.IndexCreator_OS_2_11;
+import com.rfs.worker.GlobalState;
+import com.rfs.worker.IndexRunner;
+import com.rfs.worker.MetadataRunner;
+
+public class RfsMigrateMetadata {
```

Comment:
I'm happy with this combining indices and all other metadata for now.  We'll want to make sure that users can run a migration for just one plugin configuration, one index, one component, or sets of those.  It's fine if it takes multiple runs - as long as it can be easily scripted.  I fear that slicing metadata up into every kind of resource will be cognitive overload.  All metadata shares a really similar set of requirements and similar workflow, so I think it makes sense from a technical perspective.  Compared to the completely denormalized form to do everything, this current approach should be reasonable to a customer too.